lattice_bscan_spi.py

```diff
@@ -89,12 +90,12 @@ def _detect_jtag_state(self, module):
             #   it is implied chain 1 is selected until TAP enters TLR state again
             with module.FSM():
                 with module.State("IDLE"):
-                    with module.If(~jtag_rst_n):
+                    with module.If(jtag_rti1):
```

Comment:
This seems to assume that the JTAG TAP will never be reset as long as the FPGA doesn't get re-initialised. Is this safe?